generators/typescript/model/type-reference-converters/src/AbstractTypeReferenceToTypeNodeConverter.ts

```diff
@@ -41,6 +71,60 @@ export abstract class AbstractTypeReferenceToTypeNodeConverter extends AbstractT
         }
     }
 
+    private createTypeRefenceForInlineNamedTypeForInlineUnion(typeName: DeclaredTypeName): ts.TypeNode {
+        return this.generateForInlineUnion(typeName);
+    }
+
+    private createTypeRefenceForInlineAliasNamedType(
+        typeName: DeclaredTypeName,
+        params: ConvertTypeReferenceParams.InlineAliasTypeParams
+    ): ts.TypeNode {
+        let name: ts.EntityName = ts.factory.createIdentifier(params.aliasTypeName);
+        switch (params.genericIn) {
+            case "list":
+                name = ts.factory.createQualifiedName(name, "Item");
+                break;
+            case "map":
+                name = ts.factory.createQualifiedName(name, "Value");
+                break;
+            case "set":
+                name = ts.factory.createQualifiedName(name, "Item");
+                break;
+            default:
+                return ts.factory.createTypeReferenceNode(this.getReferenceToNamedType(typeName, params));
+        }
+
+        return ts.factory.createTypeReferenceNode(name);
+    }
+
+    private createTypeRefenceForInlinePropertyNamedType({
+        parentTypeName,
+        propertyName,
+        genericIn
+    }: ConvertTypeReferenceParams.InlinePropertyTypeParams): ts.TypeNode {
+        let name = ts.factory.createQualifiedName(
+            ts.factory.createIdentifier(parentTypeName),
+            ts.factory.createIdentifier(propertyName)
+        );
+        switch (genericIn) {
+            case "list":
+                name = ts.factory.createQualifiedName(name, "Item");
```

Comment:
Let's refer to that same exported constants here - this might mean we place those constants elsewhere so it's easier to import from a shared location.